bindings/jaxrs/src/main/java/org/commonjava/indy/core/bind/jaxrs/ContentAccessHandler.java

```diff
@@ -172,6 +177,54 @@ public Response doDelete( final String packageType, final String type, final Str
         return doDelete( packageType, type, name, path, eventMetadata, null );
     }
 
+    public Response doDelete( final String packageType, final String type, final String name,
+                              final DemoteRequest request, EventMetadata eventMetadata )
+    {
+        setContext( PACKAGE_TYPE, packageType );
+
+        if ( !PackageTypes.contains( packageType ) )
+        {
+            ResponseBuilder builder = Response.status( 400 );
+            return builder.build();
+        }
+
+        Set<String> paths = request.getPaths();
+        if ( paths.isEmpty() )
+        {
+            ResponseBuilder builder = Response.status( 400 );
+            return builder.build();
+        }
+
+        final StoreType st = StoreType.get( type );
+        StoreKey sk = new StoreKey( packageType, st, name );
+
+        eventMetadata = eventMetadata.set( ContentManager.ENTRY_POINT_STORE, sk );
+        setContext( CONTENT_ENTRY_POINT, sk.toString() );
+
+        Response response;
+        Map<String, String> results = new HashMap<>();
+        for ( final String path : paths )
+        {
+            setContext( PATH, path );
+            try
+            {
+                final ApplicationStatus result = contentController.delete( sk, path, eventMetadata );
+                results.put( path, String.valueOf( result.code() ) );
+            }
+            catch ( final IndyWorkflowException e )
+            {
+                logger.error( String.format( "Failed to tryDelete artifact: %s from: %s. Reason: %s", path, name,
```

Comment:
IMO would be better to use sk here instead of name for better identification of the point of failure.

Reply:
See, let's use sk instead.